ui/src/app/views/workspace/component.ts

```diff
@@ -20,35 +20,35 @@ export class WorkspaceComponent implements OnInit {
 
   repositories: Repository[] = [];
   user: User;  // to detect if logged in
-  currentUrl: string;
-  cohortList = [];
+  cohortList: Cohort[] = [];
   constructor(
       private router: Router,
+      private route: ActivatedRoute,
       private userService: UserService,
       private repositoryService: RepositoryService,
       private cohortsService: CohortsService
   ) {}
-
+  // TODO: Replace arguments to getCohortsInWorkspace to read from URL.
   ngOnInit(): void {
     this.userService.getLoggedInUser().then(user => this.user = user);
     this.cohortsService
         .getCohortsInWorkspace(
-            WorkspaceComponent.DEFAULT_WORKSPACE_NS, WorkspaceComponent.DEFAULT_WORKSPACE_ID)
+            this.route.snapshot.url[1].path, this.route.snapshot.url[2].path)
```

Comment:
I think I will make it part of a separate PR, although I agree that I would much rather have a clearer way to handle the URL parsing.